apps/aecore/src/aec_consensus_hc.erl

```diff
@@ -275,15 +275,17 @@ micro_block_height_relative_previous_block(micro, MicroHeight) ->
 %% Custom state transitions
 state_pre_transform_key_node_consensus_switch(_Node, Trees) -> Trees.
 
-%% only called for key-blocks - this is the call where we set epoch and
-%% leader
+%% only called for key-blocks - this is the call where we set epoch and leader
 state_pre_transform_key_node(Node, PrevNode, Trees) ->
     Height = aec_block_insertion:node_height(Node),
     state_pre_transform_node(key, Height, PrevNode, Trees).
 
 state_pre_transform_micro_node(Height, PrevNode, Trees) ->
-    state_pre_transform_node(micro, Height, PrevNode, Trees).
+    {T, _} = state_pre_transform_node(micro, Height, PrevNode, Trees),
+    T.
 
+-spec state_pre_transform_node(_Type :: atom(), Height :: non_neg_integer(), _PrevNode :: any(), Trees :: aec_trees:trees())
+        -> {aec_trees:trees(), list()}.
 state_pre_transform_node(_Type, Height, _PrevNode, Trees) when Height < 1 ->
     %% No leader for genesis
     Trees;
```

Comment:
Shouldn't it return a 2-tuple?

Reply:
Dialyzer has also pointed that out. Nice catch!